File: 100-199/102/Solution.py
# Definition for a binary tree node.
from typing import Optional, List
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        store = []
        self.solve(root, 0, store)
        return store

    def solve(self, root: Optional[TreeNode], level: int, store: List[List[int]]) -> None:
        if(root is None):
            return
        # if store missing sub-store, add them
        if(len(store) < level + 1):
            for i in range(level + 1 - len(store)):
                store.append([])
        store[level].append(root.val)
        self.solve(root.left, level + 1, store)
        self.solve(root.right, level + 1, store)



        
        

File: 100-199/102/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
import java.util.*;

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        ArrayList<ArrayList<Integer>> store = new ArrayList<>(); 
        rec(store, root, true, 0);

    }

    private void rec(ArrayList<ArrayList<Integer>> store, TreeNode node, boolean newSubList, int counter){
        if(node == null){
            return ; 
        }
        int newCounter = counter; 

        if(newSubList){
            store.add(new ArrayList<>(Arrays.asList(node.val))); 
            newCounter ++; 
        }
        else{
            store.get(counter).add(node.val); 
        }

        if(node.left != null && node.right == null){
            rec(store, node.left, true, newCounter); 
        }
        else if(node.left != null && node.right != null){
            rec(store, node.left, true, newCounter); 
            rec(store, node.right, false, newCounter); 
        }
    }
}